### PR TITLE
Refinement hotfix

### PR DESCRIPTION
**Pull request by lkotipal**
- Unrefine all cells outside refine_radius
- Neighbour refinement/coarsening accounted for better

### CODE REVIEW

File: projects/project.cpp [hunk 1]
```diff
@@ -592,26 +592,28 @@ namespace projects {
...
-               } else if (neighborRef < refLevel) {
+               } else if (neighborRef < refLevel && !shouldRefineNeighbor) {
                   ++coarser_neighbors;
-               } else if ((P::useAlpha ? mpiGrid[neighbor]->parameters[CellParams::AMR_ALPHA] > P::alphaRefineThreshold : false) || (P::useJPerB ? neighborAlphaTwo > P::jperbRefineThreshold : false)) {
+               } else if (shouldRefineNeighbor) {
                   // If neighbor refines, 4 of its children will be this cells refined neighbors
                   refined_neighbors += 4;
-               } else if ((P::useAlpha ? mpiGrid[neighbor]->parameters[CellParams::AMR_ALPHA] < P::alphaCoarsenThreshold : true) && (P::useJPerB ? neighborAlphaTwo < P::jperbCoarsenThreshold : true)) {
+               } else if (shouldUnrefineNeighbor) {
                   ++coarser_neighbors;
```
Comment (markusbattarbee) [inline projects/project.cpp:617]: Btw doesn't this now return 4x too large a count for coarser neighbours as a result of unrefinement?

Comment (lkotipal) [reply projects/project.cpp:617]: This counts face neighbors of same refinement level so I don't see how it does.

Comment (markusbattarbee) [reply projects/project.cpp:617]: Ah, right, the `(neighborRef == refLevel)` is implicit.

**Review by markusbattarbee — APPROVED** [review]

File: projects/project.cpp [hunk 2]
```diff
@@ -592,24 +592,26 @@ namespace projects {
...
+            bool shouldRefine {(r2 < r_max2) && ((P::useAlpha1 ? cell->parameters[CellParams::AMR_ALPHA1] > P::alpha1RefineThreshold : false) || (P::useAlpha2 ? cell->parameters[CellParams::AMR_ALPHA2] > P::alpha2RefineThreshold : false))};
+            bool shouldUnrefine {(r2 > r_max2) || ((P::useAlpha1 ? cell->parameters[CellParams::AMR_ALPHA1] < P::alpha1CoarsenThreshold : true) && (P::useAlpha2 ? cell->parameters[CellParams::AMR_ALPHA2] < P::alpha2CoarsenThreshold : true))};
 
             // Finally, check neighbors
             int refined_neighbors {0};
             int coarser_neighbors {0};
             for (const auto& [neighbor, dir] : mpiGrid.get_face_neighbors_of(id)) {
                const int neighborRef = mpiGrid.get_refinement_level(neighbor);
-               if (neighborRef > refLevel) {
+               bool shouldRefineNeighbor {(r2 < r_max2) && ((P::useAlpha1 ? mpiGrid[neighbor]->parameters[CellParams::AMR_ALPHA1] > P::alpha1RefineThreshold : false) || (P::useAlpha2 ? mpiGrid[neighbor]->parameters[CellParams::AMR_ALPHA2] > P::alpha2RefineThreshold : false))};
```
Comment (markusbattarbee) [inline projects/project.cpp:605]: Hang, on, should the range check here in fact use an `r2` calculated for the neighbour?

Comment (lkotipal) [reply projects/project.cpp:605]: It should!

**Review by markusbattarbee — APPROVED** [review]

**Pull request merged**